src/gurt/telemetry.c

```diff
@@ -2887,8 +2887,11 @@ d_tm_add_ephemeral_dir(struct d_tm_node_t **node, size_t size_bytes,
 	d_tm_unlock_shmem();
 	return 0;
 fail_sync:
-	d_tm_del_ephemeral_dir(path);
-	goto fail_unlock; /* shmem will be closed/destroyed already */
+	d_tm_unlock_shmem();
+	rc = d_tm_del_ephemeral_dir(path);
+	if (unlikely(rc != 0))
+		DL_ERROR(rc, "failed to remove ephemeral dir @ %s", path);
+	goto fail; /* shmem will be closed/destroyed already */
```

Comment:
if the above sync_attached_segment_uid() call fails then d_tm_unlock_shmem() will not be called, is that a concern in that case?

In the other case that sync_attached_segment_uid() succeeds, then d_tm_unlock_shmem() will have been run by the time this code executes.

Reply:
sorry yeah I glossed over the unlock first line of this fail_sync block. Sorry about that. Looks good.